### PR TITLE
refactor query and option compilation

### PR DESCRIPTION
**Pull request by walterddr**
This is part2 follow up of #9191 

- move option parsing and query compile to common utils
- support option string parsing as well from controller rest api

### CODE REVIEW

File: pinot-common/src/main/java/org/apache/pinot/sql/parsers/SqlNodeAndOptions.java [hunk 1]
```diff
@@ -23,17 +23,25 @@
...
 
 public class SqlNodeAndOptions {
+  private String _sql;
   private final SqlNode _sqlNode;
   private final PinotSqlType _sqlType;
   // TODO: support option literals other than STRING
-  private final Map<String, String> _options;
-
+  private final Map<String, String> _queryOptions;
+  private final Map<String, String> _debugOptions;
```
Comment (walterddr) [inline pinot-common/src/main/java/org/apache/pinot/sql/parsers/SqlNodeAndOptions.java:31]: debug options will be removed once release 0.11 is cut. so we will wait until then before merging this PR.

**Review by siddharthteotia — APPROVED** [review]

File: pinot-common/src/main/java/org/apache/pinot/sql/parsers/CalciteSqlParser.java [hunk 2]
```diff
@@ -137,7 +137,8 @@ public static SqlNodeAndOptions compileToSqlNodeAndOptions(String sql)
     }
   }
 
-  public static SqlNodeAndOptions extractSqlNodeAndOptions(SqlNodeList sqlNodeList) {
+  @VisibleForTesting
+  static SqlNodeAndOptions extractSqlNodeAndOptions(String sql, SqlNodeList sqlNodeList) {
```
Comment (Jackie-Jiang) [inline pinot-common/src/main/java/org/apache/pinot/sql/parsers/CalciteSqlParser.java:141]: The new added `sql` seems unused?

File: pinot-common/src/main/java/org/apache/pinot/common/utils/request/RequestUtils.java [hunk 3]
```diff
@@ -30,13 +33,67 @@
...
+   */
+  @VisibleForTesting
+  public static void setOptions(SqlNodeAndOptions sqlNodeAndOptions, JsonNode jsonRequest) {
+    Map<String, String> queryOptions = new HashMap<>();
+    if (jsonRequest.has(CommonConstants.Broker.Request.DEBUG_OPTIONS)) {
+      Map<String, String> debugOptions = RequestUtils.getOptionsFromJson(jsonRequest,
+          CommonConstants.Broker.Request.DEBUG_OPTIONS);
+      // TODO: remove debug options after releasing 0.11.0.
+      if (!debugOptions.isEmpty()) {
+        // NOTE: Debug options are deprecated. Put all debug options into query options for backward compatibility.
```
Comment (Jackie-Jiang) [inline pinot-common/src/main/java/org/apache/pinot/common/utils/request/RequestUtils.java:71]: This is not enough. During version upgrade, server doesn't know the debug options can be read from the query options

File: pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/MultiStageBrokerRequestHandler.java [hunk 4]
```diff
@@ -133,16 +133,12 @@ private BrokerResponseNative handleRequest(long requestId, String query,
       throws Exception {
     LOGGER.debug("SQL query for request {}: {}", requestId, query);
 
+    // Parse the request
+    sqlNodeAndOptions = sqlNodeAndOptions != null ? sqlNodeAndOptions : RequestUtils.parseQuery(query, request);
```
Comment (Jackie-Jiang) [inline pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/MultiStageBrokerRequestHandler.java:137]: This needs to be moved into the try-catch

**Review by Jackie-Jiang — COMMENTED** [review]
We cannot merge this PR before 0.11 is cut. Currently it is still in release candidate branch, so I'd suggest reverting it for now for safety reason. Once `0.11` is officially released, we can re-post the commit

**Pull request merged**